### PR TITLE
7702 changes for devnet 5

### PR DESCRIPTION
**Pull request by ak88**
**DO NOT MERGE!**

Mekong does not include devnet 5 changes, so we cannot merge to master for now. 

All EXTCODE ops now only reads the first two bytes of the delegation header, instead of reading from delegation target. 

Changes described here:
https://github.com/ethereum/EIPs/pull/8969/files

### CODE REVIEW

File: src/Nethermind/Nethermind.Evm/VirtualMachine.cs
[the remark this answers is not in the document]
Comment (MarekM25) [reply src/Nethermind/Nethermind.Evm/VirtualMachine.cs:1]: I am almost 100% sure that this is okay though :)

**Review by MarekM25 — APPROVED** [review]
Looks good, merging it will break mekong, so let's discuss next week how do we handle devnet-5 changes

**Review by rjnrohit — APPROVED** [review]

File: src/Nethermind/Nethermind.Evm/VirtualMachine.cs
[the remark this answers is not in the document]
Comment (ak88) [reply src/Nethermind/Nethermind.Evm/VirtualMachine.cs:1]: The change only affected the "code reading" ops, but the executing ones (CALL, CALLCODE...) remains the same as before. So they are still affected by 7702 itself, but not by this new change.

File: src/Nethermind/Nethermind.Evm/VirtualMachine.cs [hunk 1]
```diff
@@ -2079,8 +2086,14 @@ static void ThrowOperationCanceledException() =>
...
     private void InstructionExtCodeSize<TTracingInstructions>(Address address, ref EvmStack<TTracingInstructions> stack, ICodeInfoRepository codeInfoRepository, IReleaseSpec spec) where TTracingInstructions : struct, IIsTracing
     {
-        ReadOnlyMemory<byte> accountCode = codeInfoRepository.GetCachedCodeInfo(_state, address, spec).MachineCode;
+        Address delegation;
+        ReadOnlyMemory<byte> accountCode = codeInfoRepository.GetCachedCodeInfo(_state, address, spec, out delegation).MachineCode;
         UInt256 result = (UInt256)accountCode.Span.Length;
+        if (delegation is not null)
+        {
+            //If the account has been delegated only the first two bytes of the delegation header counts as size 
+            result = 2;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Evm/VirtualMachine.cs:2095]:
```suggestion
            result = Eip7702Constants.FirstTwoBytesOfHeader.Count;
```

**Pull request merged**